libpldmresponder/fru.cpp

```diff
@@ -22,118 +26,117 @@ namespace responder
 
 constexpr auto root = "/xyz/openbmc_project/inventory/";
 
-std::optional<pldm_entity>
-    FruImpl::getEntityByObjectPath(const dbus::InterfaceMap& intfMaps)
+pldm_entity FruImpl::getEntityByObjectPath(const dbus::ObjectValueTree& objects,
+                                           const std::string& path)
 {
-    for (const auto& intfMap : intfMaps)
+    pldm_entity entity{};
+
+    auto iter = objects.find(path);
+    if (iter == objects.end())
+    {
+        return entity;
+    }
+
+    for (const auto& intfMap : iter->second)
     {
         try
         {
-            pldm_entity entity{};
             entity.entity_type = parser.getEntityType(intfMap.first);
-            return entity;
+            entity.entity_instance_num = 1;
+            break;
         }
         catch (const std::exception&)
         {
             continue;
         }
     }
 
-    return std::nullopt;
+    return entity;
 }
 
 void FruImpl::updateAssociationTree(const dbus::ObjectValueTree& objects,
                                     const std::string& path)
 {
-    if (path.find(root) == std::string::npos)
+    // eg: path =
+    // "/xyz/openbmc_project/inventory/system/chassis/motherboard/powersupply0"
+    // Get the parent class in turn and store it in a temporary vector
+    // eg: tmpObjPaths = {"/xyz/openbmc_project/inventory/system",
+    // "/xyz/openbmc_project/inventory/system/chassis/",
+    // "/xyz/openbmc_project/inventory/system/chassis/motherboard",
+    // "/xyz/openbmc_project/inventory/system/chassis/motherboard/powersupply0"}
+    std::string root = "/xyz/openbmc_project/inventory";
+    if (path.find(root + "/") == std::string::npos)
     {
         return;
     }
 
-    std::stack<std::string> tmpObjPaths{};
-    tmpObjPaths.emplace(path);
+    std::vector<std::string> tmpObjPaths{};
+    tmpObjPaths.push_back(path);
 
     auto obj = pldm::utils::findParent(path);
-    while ((obj + '/') != root)
+    while (obj != root)
     {
-        tmpObjPaths.emplace(obj);
+        tmpObjPaths.push_back(obj);
         obj = pldm::utils::findParent(obj);
     }
 
-    std::stack<std::string> tmpObj = tmpObjPaths;
-    while (!tmpObj.empty())
-    {
-        std::string s = tmpObj.top();
-        tmpObj.pop();
-    }
     // Update pldm entity to assocition tree
-    std::string prePath = tmpObjPaths.top();
-    while (!tmpObjPaths.empty())
+    int i = (int)tmpObjPaths.size() - 1;
```

Comment:
it would be good if you can use some for/while loop over vector, because for (; i >= 0; i--)
loop is not telling what operation is happening here.
FYR: [](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p3-express-intent)

Reply:
will check

Reply:
give meaningful name while iterating vector, don't use single int i , that's what that link suggesting.
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p3-express-intent